pkg/tests/tasks/observability/openshift_monitoring_test.go

```diff
@@ -0,0 +1,252 @@
+package observability
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+	"testing"
+
+	"github.com/maistra/maistra-test-tool/pkg/app"
+	"github.com/maistra/maistra-test-tool/pkg/util/check/assert"
+	"github.com/maistra/maistra-test-tool/pkg/util/curl"
+	"github.com/maistra/maistra-test-tool/pkg/util/env"
+	"github.com/maistra/maistra-test-tool/pkg/util/istio"
+	"github.com/maistra/maistra-test-tool/pkg/util/ns"
+	"github.com/maistra/maistra-test-tool/pkg/util/oc"
+	"github.com/maistra/maistra-test-tool/pkg/util/pod"
+	"github.com/maistra/maistra-test-tool/pkg/util/retry"
+	"github.com/maistra/maistra-test-tool/pkg/util/shell"
+	"github.com/maistra/maistra-test-tool/pkg/util/test"
+	"github.com/maistra/maistra-test-tool/pkg/util/version"
+)
+
+const (
+	monitoringNs             = "openshift-monitoring"
+	userWorkloadMonitoringNs = "openshift-user-workload-monitoring"
+	thanosTokenPrefix        = "prometheus-user-workload-token-"
+)
+
+// TestOpenShiftMonitoring requires OpenShift Monitoring stack to be enabled.
+// In case of running tests on CRC, you must enable the following setting:
+// crc config set enable-cluster-monitoring true
+// Make sure that you have sufficient resources to run monitoring stack - 8 CPU and 20 Gb of RAM should be enough:
+// crc config set memory 20480
+// crc config set cpus 8
+func TestOpenShiftMonitoring(t *testing.T) {
+	test.NewTest(t).Id("openshift-monitoring-integration").Groups(test.Full).Run(func(t test.TestHelper) {
+		smcpVer := env.GetSMCPVersion()
```

Comment:
If we have something from the documentation already done, we can put a t.Log here with the url of the documentation.